specification/snapshot/_types/SnapshotRepository.ts

```diff
@@ -17,22 +17,119 @@
  * under the License.
  */
 
-import { Uuid } from '@_types/common'
+import { ByteSize, Uuid } from '@_types/common'
 import { integer } from '@_types/Numeric'
+import { Base } from '@xpack/usage/types'
+import { AdditionalProperties } from '@spec_utils/behaviors'
+import { UserDefinedValue } from '@spec_utils/UserDefinedValue'
+import { Stringified } from '@spec_utils/Stringified'
 
-export class Repository {
-  type: string
+export class BaseRepository<Settings> {
```

Comment:
In the interest of avoiding generics if possible, what do you think about taking a more "Query"-like approach to these structures?

```ts
export class BaseRepository {
  uuid?: Uuid
}

export class BaseRepositorySettings {
  chunk_size?: ByteSize
  max_number_of_snapshots?: integer
  max_snapshot_bytes_per_sec?: ByteSize
  max_restore_bytes_per_sec?: ByteSize
  use_for_peer_recovery?: boolean
}

export class S3RepositorySettings extends BaseRepositorySettings {
  ...
}

export class S3Repository extends BaseRepository {
  type: 's3'
  settings: S3RepositorySettings
}

/**
 * @variants internal tag='type'
 * @non_exhaustive
 */
export type Repository = S3Repository | ...
```

Reply:
Not sure if avoiding generics should be a goal, but this approach also works. Done.